src/cards/target_arrow.ts

```diff
@@ -33,94 +34,91 @@ const spell: Spell = {
-  const arrowShootPath = findArrowPath(casterPositionAtTimeOfCast, target, underworld);
-  if (arrowShootPath === undefined) {
-    return [];
-  }
-  // Get all units between source and target for the arrow to pierce:
-  const hitTargets = (prediction ? underworld.unitsPrediction : underworld.units).filter(
-    (u) => {
-      if (!u.alive) {
-        return false;
+        if (!prediction && !globalThis.headless) {
+          const timeout = Math.max(0, timeoutToNextArrow);
+          await new Promise(resolve => setTimeout(resolve, timeout));
+          // Decrease timeout with each subsequent arrow fired to ensure that players don't have to wait too long
+          timeoutToNextArrow -= 5;
+        }
       }
-      // Note: Filter out self as the arrow shouldn't damage caster
-      if (u.id == casterId) {
-        return false;
+
+      await underworld.awaitForceMoves();
+      // - - - - - End copied from arrow.ts - - - - -
+
+      // Update targets variable to include newly added targets
+      targets = getCurrentTargets(state);
+      if (!prediction && !globalThis.headless && globalThis.predictionGraphics) {
+        const promises: Promise<void>[] = [];
+        targets.forEach(t => {
+          // Animations do not occur on headless
+          promises.push(new Promise<void>((resolve) => {
+            if (globalThis.predictionGraphics) {
+              globalThis.predictionGraphics.lineStyle(2, colors.targetingSpellGreen, 1.0)
+              playSFXKey('targetAquired');
+              globalThis.predictionGraphics.drawCircle(t.x, t.y, config.COLLISION_MESH_RADIUS);
+              // Show the targeting circle for a moment
+              setTimeout(resolve, 300);
+            }
+          }));
+        });
+        await Promise.all(promises);
+        globalThis.predictionGraphics.clear();
       }
-      const pointAtRightAngleToArrowPath = findWherePointIntersectLineSegmentAtRightAngle(u, arrowShootPath);
-      // TODO: Validate: Will this hit miniboss since their radius is larger?
-      const willBeStruckByArrow = !pointAtRightAngleToArrowPath ? false : distance(u, pointAtRightAngleToArrowPath) <= config.COLLISION_MESH_RADIUS;
-      return willBeStruckByArrow;
+      return state;
     },
-  ).sort(sortCosestTo(arrowShootPath.p1))
-
-  // Return the endPoint so the arrow will fly and hit a wall even if it doesn't hit a unit
-  return hitTargets.length ? hitTargets : [arrowShootPath.p2];
-}
+  },
+  events: {
+    onProjectileCollision: ({ unit, pickup, underworld, projectile, prediction }) => {
+      if (projectile.state) {
+        if (unit) {
+          addTarget(unit, projectile.state, underworld, prediction);
+        } else if (pickup) {
+          addTarget(pickup, projectile.state, underworld, prediction);
+        } else {
+          // Add collision location as target
+          //addTarget(projectile.pushedObject, projectile.state, underworld, prediction);
```

Comment:
I think prev functionality allowed for spawning a unit at the wall where the target arrow collided